src/im_mad/remotes/node-probes.d/pci.rb

```diff
@@ -90,6 +91,11 @@ def get_devices(filter = nil)
     end.flatten
 end
 
+def device_attr?(device, attribute)
+    addr = "0000:#{device[:bus]}:#{device[:slot]}.#{device[:function]}"
+    !`ls -l /sys/bus/pci/devices/#{addr}/ | grep #{attribute}`.empty?
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Identified a dynamic command in your code. If unvalidated user data can influence this call site, it poses a code injection risk. Malicious actors may exploit this weakness to inject and execute arbitrary and potentially malicious code. Ensure that user input is never directly influential to mitigate these security risks.